packages/components/addon/modifiers/hds-clipboard.js

```diff
@@ -0,0 +1,177 @@
+      textToCopy = text;
+    } else if (
+      // context: https://github.com/hashicorp/design-system/pull/1564
+      typeof text === 'number' ||
+      typeof text === 'bigint'
+    ) {
+      textToCopy = text.toString();
+    } else {
+      assert(
+        `\`hds-clipboard\` modifier - \`text\` argument must be a string - provided: ${typeof text}`
+      );
+    }
+  }
+  return textToCopy;
+};
+
+export const getTargetElement = (target) => {
+  let targetElement;
+  if (typeof target === 'string') {
+    targetElement = document.querySelector(target);
+
+    if (!targetElement) {
+      console.error(
+        '`hds-clipboard` modifier - `target` selector provided does not point to an existing DOM node, check your selector string',
+        targetElement
+      );
+      return;
+    }
+  } else if (target instanceof Node && target.nodeType === 1) {
+    targetElement = target;
+  } else {
+    if (target instanceof NodeList) {
+      assert(
+        '`hds-clipboard` modifier - `target` argument must be a string or a DOM node - provided: a list of DOM nodes'
+      );
+    } else {
+      assert(
+        `\`hds-clipboard\` modifier - \`target\` argument must be a string or a DOM node - provided: ${typeof target}`
+      );
+    }
+  }
+  return targetElement;
+};
+
+export const getTextToCopyFromTargetElement = (targetElement) => {
+  let textToCopy;
+  if (targetElement instanceof Node && targetElement.nodeType === 1) {
+    if (
+      targetElement instanceof HTMLInputElement || // targetElement.nodeName === 'INPUT' ||
+      targetElement instanceof HTMLTextAreaElement || // targetElement.nodeName === 'TEXTAREA' ||
+      targetElement instanceof HTMLSelectElement // targetElement.nodeName === 'SELECT'
+    ) {
+      textToCopy = targetElement.value;
+    } else {
+      // simplest approach
+      textToCopy = targetElement.innerText;
+
+      // approach based on text selection (left for backup just in case)
+      // var selection = window.getSelection();
+      // var range = document.createRange();
+      // selection.removeAllRanges();
+      // range.selectNodeContents(targetElement);
+      // selection.addRange(range);
+      // textToCopy = selection.toString();
+      // selection.removeAllRanges();
+    }
+  }
+  return textToCopy;
+};
+
+export const writeTextToClipboard = async (textToCopy) => {
+  // finally copy the text to the clipboard using the Clipboard API
+  // https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API
+  if (textToCopy) {
+    try {
+      // notice: the "clipboard-write" permission is granted automatically to pages when they are in the active tab
+      // https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/write
+      await navigator.clipboard.writeText(textToCopy);
+      // DEBUG uncomment this for easy debugging
+      // console.log('success', textToCopy);
+      return true;
+    } catch (error) {
+      // clipboard write failed
+      console.log(
```

Comment:
I'd suggest using Ember's `debug` package here: either warn or debug would work, but the advantage of them is that in a production build, their implementations are empty functions so you don't get logs in a production app. Here's the `warn` docs: https://api.emberjs.com/ember/5.2/functions/@ember%2Fdebug/warn

Reply:
OK, decided to change as you suggested. Thank you for raising the point.